src/common/reconciler.go

```diff
@@ -425,3 +423,8 @@ func (r *ServiceReconciler) handleProperties(service *opslevel.Service, registra
 		log.Info().Msgf("[%s] Successfully assigned property with definition: '%s' and value: '%s'", service.Name, *propertyInput.Definition.Alias, propertyInput.Value)
 	}
 }
+
+func ToJSON[T any](object T) string {
```

Comment:
i don't like this as it exposes a utility function that we shouldn't be (and shouldn't really have/need) - its only used in 1 place in prod code because of logging which can be changed to just use `%+v` in the format string and then the other place is in tests - which means it should be in the test files as a helper utility.

Reply:
>its only used in 1 place in prod code because of logging which can be changed to just use %+v in the format string

We have tried using `+%v` to log update inputs before, it does not work because it prints out pointer addresses. 

>i don't like this as it exposes a utility function that we shouldn't be (and shouldn't really have/need)

I think it would be really useful to be able to see the entire `ServiceUpdateInput` and not just the output of the diff. 

I am going to just json marshal this in the funtion in the code and move this function to the test file.

Reply:
ahh - i had not considered the problem of pointer addresses being useless :(